api/app/models/machine_learning.py

```diff
@@ -0,0 +1,215 @@
+""" Module for calculating the bias for a weather station use basic Machine Learning through Linear
+Regression.
+"""
+from datetime import datetime, timedelta
+from collections import defaultdict
+from typing import List
+from logging import getLogger
+from sklearn.linear_model import LinearRegression
+from scipy.interpolate import griddata
+import numpy as np
+from sqlalchemy.orm import Session
+from app.models import MODEL_VALUE_KEYS, construct_interpolated_noon_prediction
+from app.db.models import (
+    PredictionModel, PredictionModelGridSubset, ModelRunGridSubsetPrediction, HourlyActual)
+from app.db.crud import get_actuals_outer_join_with_predictions
+
+
+logger = getLogger(__name__)
+
+# Corresponding key values on HourlyActual and SampleCollection
+SAMPLE_VALUE_KEYS = ('temperature', 'relative_humidity')
+
+
+class LinearRegressionWrapper:
+    """ Class wrapping LinearRegression.
+    This class just adds in a handy boolean to indicate if this linear regression model is good to use.
+    """
+
+    def __init__(self):
+        self.model = LinearRegression()
+        self.good_model = False  # Flag if regression model is "good" (usable).
```

Comment:
I'm confused about what the purpose of this `good_model` flag is - around line 192 it looks like all models will eventually be marked as "good models"? Is this being used as a "complete" flag?

Reply:
Not all models are marked True, sometimes there isn't enough sample data, and then it stays False.

You're right in saying that "complete" is probably more appropriate, or maybe even just setting it to None if it's not working. What's actually happening, is that I removed a bit of code that was looking at the fit, and throwing away a regression model if it wasn't good. I want to get back to doing that, but we don't have enough data right now for me to make a sensible decision.

If you want, we can just drop that for now and simplify the code?

Reply:
I'm fine with keeping the code the way it is now. If you've got plans to make use of that flag in the future, might as well keep it in :)